R/zchunk_LA161.Cstorage_china.R

```diff
@@ -0,0 +1,125 @@
+    L161.Csupply_province <- Dooley_CCS_China %>%
+      # Add grid.region
+      left_join_error_no_match(province_names_mappings, by = "province.name") %>%
+      # Convert prices to 1990 USD per ton C and amounts to C
+      mutate(Cost_1990USDtC = Cost_2005USDtCO2 * emissions.CONV_C_CO2 / gdp_deflator(2005, 1990),
+             MtC = CO2_Mt * (1 / emissions.CONV_C_CO2)) %>%
+      arrange(province.name, Cost_2005USDtCO2) %>%
+    # Removing data for provinces with not enough onshore storage
+      filter(!(province.name %in% c("Fujian", "Guangdong", "Guangxi", "Qinghai")))
+
+    province_data <- L161.Csupply_province %>%
+      group_by(province.name) %>%
+      # Calculate cumulative sum, then filter to the quantiles
+      mutate(Cumul_MtC = cumsum(MtC))
+
+    L161.Cstorage_province <- province_data %>%
+      # use "type=1" to indicate that we want to return only exact matches to the data in the table.
+      # only take the (0, 0.2, 0.4, 0.6, 0.8, 1) points in the quartiles
+      filter(Cumul_MtC %in% quantile(Cumul_MtC, c(0, 0.2, 0.4, 0.6, 0.8, 1), type = 1)) %>%
+      group_by(province.name) %>%
+      mutate(# this will create grade equal to grade 1 for 0, grade 2 for 0.2, etc.
+             grade = paste("grade",1:6, sep = " "),
+             # from the cumulative totals, return to the marginal quantities associated with each grade
+             MtC = Cumul_MtC- lag(Cumul_MtC,1,0)) %>%
+      ungroup %>%
+      # Return from the list to a data frame with all necessary data
+      select(province.name, grade, Cost_1990USDtC, MtC) %>%
+      # Add cost grades to provinces without data
+      bind_rows(CCS_China_add) %>%
+      mutate(# Setting a minimum cost of 0 on CO2 storage and transport projects
+             Cost_1990USDtC = pmax(Cost_1990USDtC, 0))
+
+    # Manually changing cost curves to avoid grades 2 or 3 being 0. Value determined by linear interpolation. If the data changes in the future, will need to change this
+    L161.Cstorage_province$Cost_1990USDtC[which(L161.Cstorage_province$province.name== "Jilin" & L161.Cstorage_province$grade== "grade 2")]    <- 4.625
+    L161.Cstorage_province$Cost_1990USDtC[which(L161.Cstorage_province$province.name== "Xinjiang" & L161.Cstorage_province$grade== "grade 2")] <- 4.3
+    L161.Cstorage_province$Cost_1990USDtC[which(L161.Cstorage_province$province.name== "Yunnan" & L161.Cstorage_province$grade== "grade 2")]   <- 5.74
+    L161.Cstorage_province$Cost_1990USDtC[which(L161.Cstorage_province$province.name== "Yunnan" & L161.Cstorage_province$grade== "grade 3")]   <- 11.48
+
+    #Extending the plateau (bin 2) of the cost curve using additional resource data
+    L161.Cstorage_province_g2 <- L161.Cstorage_province %>%
+      filter(grade == "grade 2") ->
+      L161.Cstorage_province_g2
+
+    L161.Cstorage_province_g2 %>%
+      mutate(add_storage = left_join_error_no_match(L161.Cstorage_province_g2, Unallocated_CStorage, by = c("province.name" = "province.name"))$MtCO2,
```

Comment:
`left_join_error_no_match` first, then mutate, do not combine the two

Reply:
What happens here:
1. Add a new column named as add_storage
2. The default value of add_storage comes from MtCO2 of Unallocated_CStorage

I don't how to add a new column without using mutate, so...